cpp/src/arrow/flight/serialization_internal.cc

```diff
@@ -201,9 +193,7 @@ grpc::Status FlightDataSerialize(const FlightPayload& msg, ByteBuffer* out,
   // Write the descriptor if present
   int32_t descriptor_size = 0;
   if (msg.descriptor != nullptr) {
-    if (msg.descriptor->size() > kInt32Max) {
-      return ToGrpcStatus(Status::CapacityError("Descriptor size overflow (>= 2**31)"));
-    }
+    DCHECK_LT(msg.descriptor->size(), kInt32Max);
```

Comment:
It's a bit strange to continue running here, if we know it will fail finally (will it?)

From the jira issue, looks grpc should check and handle our return value decently.
Maybe open an issue in grpc community? If grpc is willing to change, do we still need to update our code?

For now, I'm not sure if it's appropriate to simply abort here in release build, with some clear error messages and hints to fix it by chunking the data.

Reply:
Whoops, fixed (here and elsewhere), thanks.